woocommerce/woocommerce-android
# My store using timestamp last synced

⚠️  Warning: This PR is a part of a more extensive work, please do not merge

Closes: #9443

### Why
As part of the Reliability project, we are working on improving the app's performance and reducing the perceived time it takes to load. One of the ideas we are working on is introducing a timestamp as the web does and preventing redundant requests to the API. This PR is part of the timestamp introduction in the My Store screen.

This how the web looks like:
<img width="594" alt="Screenshot 2023-06-21 at 11 25 43" src="https://github.com/woocommerce/woocommerce-android/assets/18119390/133b162b-f9f9-4344-bdb1-577f07cbbb43">

### Description
This PR changes the logic followed by the My Stats screen on where a request was forced to the API and follows the same strategy as the Analytics Hub screen. With these changes, the two screens (Hub and My Store) will share the timestamp for the last update and rely more on the cached data. 

### Testing instructions
TC1
1. Open the App and open a granularity (ex. Today)
2. Wait until the screen finish loading
3. Press on a different granularity (ex. This week)
4.  Wait until the screen finish loading
5. Change back to the first selection (Today)
6. Check that no request is made to the API and that data is presented from the cache
7. Change to the second selection (This week)
8. Check that no request is made to the API and that data is presented from the cache

TC2
1. Open the App and open a granularity (ex. Today)
2. Wait until the screen finish loading
3. Press on a different granularity (ex. This week)
4. Wait until the screen finish loading
5. Change back to the first selection (Today)
6. Check that no request is made to the API and that data is presented from the cache
7. Pull-To-Refresh and check that a call is made to the API and the data is refreshed

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/5189794a-8379-442f-895d-27d8e1cbf773
